packages/mds-agency/request-handlers.ts

```diff
@@ -330,16 +320,32 @@ export const submitVehicleEvent = async (req: AgencyApiRequest, res: AgencyApiRe
 
     // database write is crucial; failures of cache/stream should be noted and repaired
     const recorded_event = await db.writeEvent(event)
+    const { telemetry } = recorded_event
+
+    if (telemetry) {
+      await db.writeTelemetry(Array.isArray(telemetry) ? telemetry : [telemetry])
+    }
+
     try {
       await Promise.all([
         cache.writeEvent(recorded_event),
         stream.writeEvent(recorded_event),
         socket.writeEvent(recorded_event)
       ])
-      await finish()
+
+      if (telemetry) {
```

Comment:
Will this get weird with `[]`?

Reply:
Not really, I suppose it’ll marginally increase runtime because we’re not short circuiting, but all of the nested logic can handle an empty array.